protocol-units/execution/executor/src/lib.rs

```diff
@@ -13,9 +13,10 @@ use aptos_api::runtime::Apis;
 use movement_types::BlockCommitment;
 
 use async_channel::Sender;
+use async_trait::async_trait;
 
-#[tonic::async_trait]
-pub trait SuzukaExecutor {
+#[async_trait]
+pub trait Executor {
 
```

Comment:
Idk if you are trying to place this in this PR or save it for another one, but I don think the design we discussed is probably more readable. 

Either have separate methods for executing each `dyn`, `opt`, and `fin` block type or having one `execute_block` that demuxes an enum of those different blocks, would be ideal IMO.

Reply:
We will only execute `opt` blocks, though `dyn` might come later. `fin` is updated with `set_finalized_block_height`.

I don't see value in multiplexing these requests in a single method at this point, the way integration will be done in the full node. We could design a channel for sending these events to the executor task that runs in the background, but it's something that should be left for the general rearchitecting called for in #63.